src/core/tsi/s2a/handshaker/s2a_handshaker_client.h

```diff
@@ -19,62 +19,164 @@
 #ifndef GRPC_CORE_TSI_S2A_HANDSHAKER_S2A_HANDSHAKER_CLIENT_H
 #define GRPC_CORE_TSI_S2A_HANDSHAKER_S2A_HANDSHAKER_CLIENT_H
 
+#include <grpc/support/port_platform.h>
+
 #include <grpc/byte_buffer.h>
 #include <grpc/byte_buffer_reader.h>
 #include <grpc/grpc.h>
+
 #include "src/core/lib/iomgr/closure.h"
 #include "src/core/lib/iomgr/pollset_set.h"
+#include "src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h"
 #include "src/core/tsi/transport_security_interface.h"
 
+using ::experimental::grpc_s2a_credentials_options;
+
 namespace grpc_core {
 namespace experimental {
 
-typedef struct s2a_tsi_handshaker s2a_tsi_handshaker;
+struct s2a_tsi_handshaker;
+
+/** A function that makes a gRPC call to the S2A. The default caller option
+ *  is the grpc_call_start_batch_and_execute caller. **/
+typedef grpc_call_error (*s2a_grpc_caller)(grpc_call* call, const grpc_op* ops,
+                                           size_t nops, grpc_closure* tag);
+
+/** A struct that stores the handshake result sent by the S2A service. **/
+struct s2a_recv_message_result {
+  tsi_result status;
+  const uint8_t* bytes_to_send;
+  size_t bytes_to_send_size;
+  tsi_handshaker_result* result;
+};
 
 /** The Secure Session Agent (S2A) handshaker client interface. It facilitates
  *  establishing a secure channel with the peer by interacting with the
  *  S2A's handshaker service. More precisely, it schedules
  *  a handshaker request that could be one of client_start, server_start,
  *  and next handshaker requests. The interface and all API's are
  *  thread-compatible. **/
-typedef struct s2a_handshaker_client s2a_handshaker_client;
+struct s2a_handshaker_client {
```

Comment:
Why not make it a class? Please also rename the functions appropriately.

Reply:
Done in next commit.